src/k8s-engine/wsl.ts

```diff
@@ -1102,16 +1104,41 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
         })(),
         ];
 
+        await Promise.all(downloadingActions);
         if (config.enabled) {
           desiredVersion = await this.desiredVersion;
-          downloadingActions.push(
-            this.progressTracker.action(
-              'Checking k3s images',
-              100,
-              this.k3sHelper.ensureK3sImages(desiredVersion)),
-          );
+          try {
+            await this.progressTracker.action('Checking k3s images', 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          } catch (ex:any) {
+            console.log(`Failed to find version ${ desiredVersion.raw }: ${ ex }`, ex);
+            if (ex.code === 'ECONNREFUSED' || ex.toString().includes('getaddrinfo ENOTFOUND github.com')) {
```

Comment:
Is that the actual message on Windows? It feels very Unix-y.
(See also the lima code — this should be 1) in the `desiredVersion` getter, and 2) maybe get factored out to something else to detect transient network errors.)

Reply:
The message is unixy because it's coming from `kuberlr`. I'll go with helper functions